### PR TITLE
[dev-v2.7] Add K8s September patches

### PR DESCRIPTION
**Pull request by rayandas**
This PR includes K8s September patches ([issue](https://github.com/rancher/rancher/issues/38828)), and fix for metrics server 0.5.0 failing due to docker CIS hardening ([issue](https://github.com/rancher/rancher/issues/37631))

### CODE REVIEW

File: rke/templates/metrics_v0.5.0.go
Comment (snasovich) [inline rke/templates/metrics_v0.5.0.go]: We need to introduce a new template as per discussions in this thread - https://github.com/rancher/kontainer-driver-metadata/pull/938/files#r953052960

**Review by snasovich — COMMENTED** [review]
See added comment, we need a new template to properly fix metrics server issue.
Not using "Request changes" so merging is not blocked in case I'm not around to clear the reject. But these changes need to be addressed.

File: rke/templates/metrics_v0.5.0.go
Comment (rayandas) [reply rke/templates/metrics_v0.5.0.go]: Alright! Few questions. The next template verion should be `metrics_v0.5.1.go` ? And what K8s version should support this new metrics version? The current version has `">=1.20.14-rancher2-1 <1.23.3-rancher1-1": metricsServerv050` see https://github.com/rancher/kontainer-driver-metadata/blob/dev-v2.7/rke/templates/templates.go#L160
cc: @kinarashah

Comment (rayandas) [reply rke/templates/metrics_v0.5.0.go]: @kinarashah @snasovich I added a template metrics_v0.5.1.go also updated the range of K8s supported version for metric server
```
">=1.22.14-rancher1-1 <1.23.3-rancher1-1": metricsServerv051,
">=1.20.14-rancher2-1 <1.22.13-rancher1-1": metricsServerv050,
```
Please review and let me know what all changes are required.

File: rke/templates/metrics_v0.5.1.go
Comment (snasovich) [inline rke/templates/metrics_v0.5.1.go]: The chosen file name is confusing as it makes impression that it's for metrics server 0.5.1.
But in fact we've been using `rancher/mirrored-metrics-server:v0.5.1` with `metrics_v0.5.0.go` template for a while now.
Maybe name it (and related variables) as `metrics_v0.5.1-fix.go` to help avoid confusion? @kinarashah , please chime in on this.

File: rke/templates/templates.go
Comment (snasovich) [inline rke/templates/templates.go]: These ranges exclude `1.22.13-rancher1-1` altogether meaning that there is no template for `1.22.13-rancher1-1` and it would get broken.  Also, we're no longer introducing `1.22.14-rancher1-1` so I recommend the following instead:
```suggestion
			">=1.22.15-rancher1-1 <1.23.3-rancher1-1": metricsServerv051,
			">=1.20.14-rancher2-1 <=1.22.13-rancher1-1": metricsServerv050,
```

Comment (snasovich) [inline rke/templates/templates.go]: Also, key and values are not aligned in this block (need more spaces for proper indentation)

**Review by snasovich — CHANGES_REQUESTED** [review]
Some comments to address, otherwise LGTM

File: rke/templates/metrics_v0.5.1.go
Comment (rayandas) [reply rke/templates/metrics_v0.5.1.go]: Sure. Will be updating it.

File: rke/templates/templates.go
Comment (rayandas) [reply rke/templates/templates.go]: Ah! right. Thanks for the catch.

File: rke/templates/templates.go
Comment (snasovich) [inline rke/templates/templates.go]: I've just realized that all other template version ranges use exclusive top end of range, so _ideally_ we want to be consistent and have it as below:
```suggestion
			">=1.20.14-rancher2-1 <1.22.15-rancher1-1": metricsServerv050,
```

However this is a nitpick plus I was the one who suggested inclusive top end of the range, so approving.

If you have a chance to fix it before the second approval is done, that would be great. If this is the only change, consider my approval still valid even if GitHub automatically removes it.

**Review by snasovich — DISMISSED** [review]
LGTM, one minor comment that's may be addressed depending on timing.

File: rke/templates/templates.go
Comment (rayandas) [reply rke/templates/templates.go]: I've updated your suggestion.

**Review by kinarashah — APPROVED** [review]
Changes LGTM, 
- confirmed we need a new metrics-server template because latest v0.6.1 has more changes that 0.5.x don't use 
- no need for service options, version info looks right 
- default k8s is updated

**Comment by kinarashah** [comment]
@rayandas If possible, cherry-pick these commits to the custom branch created for 2.6 in the same PR so it's easy to backport and review.

**Comment by rayandas** [comment]
Yeah I will cherry-pick these commits to the backport PR and update. 
Merging this PR.

**Pull request merged**
